Merging Rust STD side of std-fs integration

CPTforever opened the pull request:
I've added some code that will interface with the Twizzler runtime. This allows user programs to use the Rust std fs interface to access Twizzler objects.

dbittman commented on library/std/src/sys/hermit/fd.rs:
```diff
@@ -26,7 +26,7 @@ impl FileDesc {
 
     pub fn write(&self, buf: &[u8]) -> io::Result<usize> {
-        let result = cvt(unsafe { abi::write(self.fd.as_raw_fd(), buf.as_ptr(), buf.len()) })?;
+        let result = cvt(unsafe { abi::write(self.fd.as_raw_fd(), buf.as_mut_ptr(), buf.len()) })?;
```
Why is there a change in the hermit code?

dbittman commented on library/std/src/sys/twizzler/fs.rs:
```diff
@@ -0,0 +1,371 @@
+    }
+
+    pub fn open_c(path: &CStr, opts: &OpenOptions) -> io::Result<File> {
+        println!("opening 1");
```
We should try to keep the code free of these indicators.

dbittman commented on library/std/src/sys/twizzler/fs.rs:
```diff
@@ -0,0 +1,371 @@
+
+impl DirBuilder {
+    pub fn new() -> DirBuilder {
+        println!("Inited DirBuilder! :)");
```
Here, too

dbittman commented on library/std/src/sys/twizzler/fs.rs:
```diff
@@ -0,0 +1,371 @@
+
+impl fmt::Debug for File {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "No")
```
This should do something better -- write the fd number or something?

dbittman commented on library/std/src/sys/twizzler/mod.rs:
```diff
@@ -6,7 +6,8 @@ pub mod args;
 pub mod cmath;
 pub mod env;
-#[path = "../unsupported/fs.rs"]
+//#[path = "../unsupported/fs.rs"]
```
Remove the commented #[attr]

dbittman reviewed (changes requested):
Just some small changes

CPTforever commented:
Undid change to hermit 
Removed all print statements
Removed commented $[attr]
Added a display trait to FileDesc and a debug trait to FIle so it should print out the file descriptor number
